driver-core/src/test/functional/com/mongodb/internal/connection/DefaultConnectionPoolTest.java

```diff
@@ -443,6 +443,41 @@ public void checkoutHandOverMechanism() throws InterruptedException, TimeoutExce
         }
     }
 
+    @Test
+    public void invalidateAfterCloseMustNotThrow() {
+        provider = new DefaultConnectionPool(
+                SERVER_ID,
+                connectionFactory,
+                ConnectionPoolSettings.builder().maxSize(1).build(),
+                mockSdamProvider());
+        provider.ready();
+        provider.close();
+        provider.invalidate(null);
+    }
+
+    @Test
+    public void invalidateConcurrentWithCloseMustNotThrow() throws ExecutionException, InterruptedException {
+        Future<?> backgroundTask = null;
+        for (int i = 0; i < 10_000; i++) {
+            provider = new DefaultConnectionPool(
+                    SERVER_ID,
+                    connectionFactory,
+                    ConnectionPoolSettings.builder().maxSize(1).build(),
+                    mockSdamProvider());
+            try {
+                backgroundTask = cachedExecutor.submit(() -> {
+                    provider.ready();
+                    provider.invalidate(null);
```

Comment:
Does this line just increase the likelihood of failure in the case of a regression?  I was able to get the test to fail (after commenting out the bug fix, that is) after one iteration even with it commented out.

Reply:
This test checks that nether `ready` nor `invalidate` throw when called concurrently with `close`. Renamed the test to `readyInvalidateConcurrentWithCloseMustNotThrow`.